x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/azureopenai/AzureOpenAiEmbeddingsRequest.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.message.BasicHeader;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.inference.common.Truncator;
+import org.elasticsearch.xpack.inference.external.azureopenai.AzureOpenAiAccount;
+import org.elasticsearch.xpack.inference.external.request.HttpRequest;
+import org.elasticsearch.xpack.inference.external.request.Request;
+import org.elasticsearch.xpack.inference.services.azureopenai.embeddings.AzureOpenAiEmbeddingsModel;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.external.request.RequestUtils.createAuthBearerHeader;
+import static org.elasticsearch.xpack.inference.external.request.azureopenai.AzureOpenAiUtils.API_KEY_HEADER;
+
+public class AzureOpenAiEmbeddingsRequest implements AzureOpenAiRequest {
+    private final Truncator truncator;
+    private final AzureOpenAiAccount account;
+    private final Truncator.TruncationResult truncationResult;
+    private final URI uri;
+    private final AzureOpenAiEmbeddingsModel model;
+
+    public AzureOpenAiEmbeddingsRequest(
+        Truncator truncator,
+        AzureOpenAiAccount account,
+        Truncator.TruncationResult input,
+        AzureOpenAiEmbeddingsModel model
+    ) {
+        this.truncator = Objects.requireNonNull(truncator);
+        this.account = Objects.requireNonNull(account);
+        this.truncationResult = Objects.requireNonNull(input);
+        this.model = Objects.requireNonNull(model);
+        this.uri = model.getUri();
+    }
+
+    public HttpRequest createHttpRequest() {
+        HttpPost httpPost = new HttpPost(uri);
+
+        String requestEntity = Strings.toString(
+            new AzureOpenAiEmbeddingsRequestEntity(
+                truncationResult.input(),
+                model.getTaskSettings().user(),
+                model.getServiceSettings().dimensions(),
+                model.getServiceSettings().dimensionsSetByUser(),
+                model.getServiceSettings().encodingFormat(),
+                model.getServiceSettings().encodingFormatSetByUser()
+            )
+        );
+
+        ByteArrayEntity byteEntity = new ByteArrayEntity(requestEntity.getBytes(StandardCharsets.UTF_8));
+        httpPost.setEntity(byteEntity);
+
+        httpPost.setHeader(new BasicHeader(HttpHeaders.CONTENT_TYPE, XContentType.JSON.mediaType()));
+
+        var entraId = model.getSecretSettings().entraId();
+        var apiKey = model.getSecretSettings().apiKey();
+
+        if (entraId != null && entraId.isEmpty() == false) {
+            httpPost.setHeader(createAuthBearerHeader(entraId));
+        } else if (apiKey != null && apiKey.isEmpty() == false) {
+            httpPost.setHeader(new BasicHeader(API_KEY_HEADER, apiKey.toString()));
+        }
```

Comment:
In theory it shouldn't happen as you perform null-checks at other places, but maybe it's worth to add an `else` execution path to guard against this error here, too, if something changes somewhere else?